packages/core/src/ui/widgets/colorBar/ViewportColorBar.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  eventTarget,
+  VolumeViewport,
+  StackViewport,
+  Types,
+  Enums,
+  utilities,
+  getEnabledElement,
+  cache,
+} from '@cornerstonejs/core';
+import { ColorBar } from './ColorBar';
+import { ColorBarProps, ColorBarVOIRange } from './types';
+
+const { Events } = Enums;
+const DEFAULT_MULTIPLIER = 4;
+
+export interface ViewportColorBarProps extends ColorBarProps {
+  element: HTMLDivElement;
+  volumeId?: string;
+}
+
+class ViewportColorBar extends ColorBar {
+  private _element: HTMLDivElement;
+  private _volumeId: string;
+
+  constructor(props: ViewportColorBarProps) {
+    super({
+      ...props,
+      range: ViewportColorBar._getRange(props.element, props.volumeId),
+      voiRange: ViewportColorBar._getVOIRange(props.element, props.volumeId),
+    });
+
+    this._element = props.element;
+    this._volumeId = props.volumeId;
+
+    this._addCornerstoneEventListener();
+  }
+
+  public get element() {
+    return this._element;
+  }
+
+  public get enabledElement() {
+    return getEnabledElement(this._element);
+  }
+
+  protected getVOIMultipliers(): [number, number] {
+    const { viewport } = this.enabledElement;
+    // const {actor: volumeActor } = viewport.getActor(this._volumeId)
+    const volume = cache.getVolume(this._volumeId);
+    const { scaling } = volume;
+    const isPreScaled = !!scaling && Object.keys(scaling).length > 0;
+    const { Modality: modality } = volume.metadata;
+
+    if (modality === 'PT') {
+      const ptMultiplier =
+        5 / Math.max(this.containerSize.width, this.containerSize.height);
+
+      return isPreScaled ? [0, ptMultiplier] : [0, DEFAULT_MULTIPLIER];
+    }
+
+    return [DEFAULT_MULTIPLIER, DEFAULT_MULTIPLIER];
+  }
+
+  protected voiChanged(voiRange: ColorBarVOIRange) {
+    super.voiChanged(voiRange);
+
+    const { viewport } = this.enabledElement;
+
+    if (viewport instanceof StackViewport) {
+      viewport.setProperties({
+        voiRange: voiRange,
+      });
+      viewport.render();
+    } else if (viewport instanceof VolumeViewport) {
+      const { _volumeId: volumeId } = this;
+      const viewportsContainingVolumeUID = utilities.getViewportsWithVolumeId(
+        volumeId,
+        viewport.renderingEngineId
+      );
+
+      viewport.setProperties({ voiRange }, volumeId);
+      viewportsContainingVolumeUID.forEach((vp) => vp.render());
+    }
+  }
+
+  private static _getRange(element, volumeId) {
```

Comment:
let's move away from getRange, and range to getImageDataRange and imageDataRange so that is more clear vs voiRange